sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/interpreter/operator/BeamSqlPrimitive.java

```diff
@@ -111,54 +112,57 @@ public boolean accept() {
     }
 
     switch (outputType) {
-    case BIGINT:
-      return value instanceof Long;
-    case DECIMAL:
-      return value instanceof BigDecimal;
-    case DOUBLE:
-      return value instanceof Double;
-    case FLOAT:
-      return value instanceof Float;
-    case INTEGER:
-      return value instanceof Integer;
-    case SMALLINT:
-      return value instanceof Short;
-    case TINYINT:
-      return value instanceof Byte;
-    case BOOLEAN:
-      return value instanceof Boolean;
-    case CHAR:
-    case VARCHAR:
-      return value instanceof String || value instanceof NlsString;
-    case TIME:
-      return value instanceof ReadableInstant;
-    case TIMESTAMP:
-    case DATE:
-      return value instanceof ReadableInstant;
-    case INTERVAL_SECOND:
-    case INTERVAL_MINUTE:
-    case INTERVAL_HOUR:
-    case INTERVAL_DAY:
-    case INTERVAL_MONTH:
-    case INTERVAL_YEAR:
-      return value instanceof BigDecimal;
-    case SYMBOL:
-      // for SYMBOL, it supports anything...
-      return true;
-    case ARRAY:
-      return value instanceof List;
-    case MAP:
-      return value instanceof Map;
-    case ROW:
-      return value instanceof Row;
-    default:
-      throw new UnsupportedOperationException(
-          "Unsupported Beam SQL type in expression: " + outputType.name());
+      case BIGINT:
+        return value instanceof Long;
+      case DECIMAL:
+        return value instanceof BigDecimal;
+      case DOUBLE:
+        return value instanceof Double;
+      case FLOAT:
+        return value instanceof Float;
+      case INTEGER:
+        return value instanceof Integer;
+      case SMALLINT:
+        return value instanceof Short;
+      case TINYINT:
+        return value instanceof Byte;
+      case BOOLEAN:
+        return value instanceof Boolean;
+      case CHAR:
+      case VARCHAR:
+        return value instanceof String || value instanceof NlsString;
+      case TIME:
+        return value instanceof ReadableInstant;
+      case TIMESTAMP:
+      case DATE:
+        return value instanceof ReadableInstant;
+      case INTERVAL_SECOND:
+      case INTERVAL_MINUTE:
+      case INTERVAL_HOUR:
+      case INTERVAL_DAY:
+      case INTERVAL_MONTH:
+      case INTERVAL_YEAR:
+        return value instanceof BigDecimal;
+      case SYMBOL:
+        // for SYMBOL, it supports anything...
+        return true;
+      case ARRAY:
+        return value instanceof List;
+      case MAP:
+        return value instanceof Map;
+      case ROW:
+        return value instanceof Row;
+      case MULTISET:
+        return value instanceof Iterable;
```

Comment:
this says that Multiset can be represented by any iterable, is this correct?

Reply:
Yes, I believe that is correct. It seems to make semantic sense - anything where you can get the values out of it, basically. And MULTISET is actually the least constrained of any collection (am I right?) so I think anything works.